base/bucket_n1ql.go

```diff
@@ -20,7 +20,8 @@ const PrimaryIndexName = "#primary"
 
 // IndexOptions used to build the 'with' clause
 type N1qlIndexOptions struct {
-	NumReplica      uint `json:"num_replica,omitempty"`          // Number of replicas
-	IndexTombstones bool `json:"retain_deleted_xattr,omitempty"` // Whether system xattrs on tombstones should be indexed
-	DeferBuild      bool `json:"defer_build,omitempty"`          // Whether to defer initial build of index (requires a subsequent BUILD INDEX invocation)
+	NumReplica      uint    `json:"num_replica,omitempty"`          // Number of replicas
+	IndexTombstones bool    `json:"retain_deleted_xattr,omitempty"` // Whether system xattrs on tombstones should be indexed
+	DeferBuild      bool    `json:"defer_build,omitempty"`          // Whether to defer initial build of index (requires a subsequent BUILD INDEX invocation)
+	NumPartitions   *uint32 `json:"num_partition"`                  // The number of partitions to use for the index. 1 will be a non-partitioned index.
```

Comment:
I think we want omitempty here to avoid sending {"num_partition":null} when NumPartitions isn't defined?